src/index.cpp

```diff
@@ -2479,9 +2306,9 @@ inline void Index<T, TagT, LabelT>::process_delete(const tsl::robin_set<uint32_t
         if (expanded_nodes_set.size() <= range)
         {
             std::unique_lock<non_recursive_mutex> adj_list_lock(_locks[loc]);
-            _final_graph[loc].clear();
+            _graph_store->clear_neighbours((location_t)loc);
             for (auto &ngh : expanded_nodes_set)
-                _final_graph[loc].push_back(ngh);
+                _graph_store->add_neighbour((location_t)loc, ngh);
```

Comment:
instead of adding one by one, can't we use set_nbrs method?

Reply:
same thing happens in 1380 also.

Reply:
This is a good catch.  I have updated all instances where we can directly call set neighbours()